### ISSUE TITLE
How can i remove the added [] from POST array data?

### ISSUE THREAD

**Issue by savokiss** [vuejs/vue-resource#407]
Noticed that this line [url/index/serialize function](https://github.com/vuejs/vue-resource/blame/master/src/url/index.js#L116) add the `[]` directly to the array data in the params, but with some backend language like java, it default not support array data with `[]` but with dulplicate names like:
form data: 
```
//This will not work:
id: 1,
gids[]: 123,
gids[]: 234
```

```
//This will work:
id: 1,
gids: 123,
gids: 234
```

So how about make `[]` a option to configure just like [ng-file-upload](https://github.com/danialfarid/ng-file-upload#upload-service) using `arrayKey: '[i]' or '[]' or '.i' or '' //default is '[i]'` to adapt to different server implementations ?

**Comment by Smallpath** [comment]
You can use `$resource` to convert query arrays.

```
  return vue.$resource('/proxyPrefix/api/post{?keys,values}').get({
    keys:['foo','bar'],
    values: ['1','2']
  }).then((response) => {
    return response.body[0];
  }, (err) => {
    console.log(err)
  })
```
Code above will return `/proxyPrefix/api/post?keys=foo,bar&values=1,2`

See the document [here](https://github.com/vuejs/vue-resource/blob/master/docs/resource.md)

**Comment by savokiss** [comment]
@Smallpath Sorry, I don't use `Vue.resource` but `Vue.http`, and just need the simple post method to support array data with no `[]`, any clue?

**Comment by Smallpath** [comment]
@savokiss As far as I know, any input uri will be converted according to `URI Template`.  I didn't find any other solution except `Vue.resource`. Actually, it's the problem of query string, not Method such as `POST` or `GET`.  

If you just want to post data to server, you should send the data as body, not param.

**Comment by steffans** [comment]
[URI Templating](https://medialize.github.io/URI.js/uri-template.html) works the same for `Vue.http`, using the `params` option try this:

```js
  Vue.http.get('/someUrl/{?keys,values}', {
     params: {keys:['foo','bar'], values: ['1','2']}
  }).then((res) => {
    // success callback
  }, (err) => {
    // error callback
  })
```

**Comment by savokiss** [comment]
@Smallpath @steffans My fault, I think there were some misunderstanding...I do mean array data in the post body.

My code like this :
```
Vue.http.post('/someUrl', {
    id: 1,
    gids: [123, 456]
}
```
But you can see the post request in chrome's network tab as below:
```
//form data
id: 1,
gids[]: 123,
gids[]: 456
```
Not my expected like this:
```
//form data
id: 1,
gids: 123,
gids: 456
```

So is there a way to let the post data act like the last one?

**Comment by Smallpath** [comment]
Just set the `Content-type` to `application/json` in the request headers. I don't remember clearly, but `vue-resource` has an option for this, [via](https://github.com/vuejs/vue-resource/blob/master/docs/config.md#legacy-web-servers)

**Comment by savokiss** [comment]
@Smallpath No, that doesn't work. That will make the `Form Data` a JSON string but `key/value` pairs.

**Comment by LancerComet** [comment]
This confuse me too, did you get any solution for vue-resource ? :(

**Comment by Doeke** [comment]
This is also an issue for me, why does it append `[]` to POST fields? I think it's a non-standard Php thing?

Is `array.toString` before submit the only way to fix this?

**Comment by shipengyan** [comment]
@savokiss  , you can use `post` method directly. In the backend server (eg, springMVC) , you can use `@RequestBody`， get param info.
![qq 20170421155811](https://cloud.githubusercontent.com/assets/1212932/25268453/5c691c54-26ab-11e7-92b4-d3f5c1b0139c.png)

**Comment by mqli** [comment]
Something like the ` jQuery.ajaxSettings.traditional` will be really helpful, since the ` []`  param may not be supported by some backend framework.
[ http://api.jquery.com/jquery.param/]( http://api.jquery.com/jquery.param/)

**Comment by italomandara** [comment]
this seems the line that causes the problem
https://github.com/pagekit/vue-resource/blob/develop/src/url/index.js#L124
but apparently there is no option to disable it, if an array or object is passed you'll get that '[]' after the parameter name

**Comment by qianfangyuan** [comment]
@smallpath   does this have  option to fix post array without prefix [] ?
 I also have problem in go.
